File: backend/CarbonWise.Api/Entities/Goal.cs
namespace CarbonWise.Api.Entities;

public class Goal
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public string GoalType { get; set; } = string.Empty;

    public decimal TargetValue { get; set; }

    public decimal CurrentValue { get; set; }

    public string Status { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; }

    public User User { get; set; } = null!;
}

File: frontend/src/types/goal.ts
export interface Goal {
  id: number;
  goalType: string;
  targetValue: number;
  currentValue: number;
  status: string;
  createdAt?: string;
}

export interface GoalTarget {
  id: number;
  goalType: string;
  targetReductionPercentage: number;
  targetDate: string;
  status: string;
}
